arcgis-ios-sdk-samples/Layers/Group layers/LayersTableViewController.swift

```diff
@@ -0,0 +1,156 @@
+        }
+        
+        guard let groupLayer = getLayer(from: layers, at: indexPath.section) as? AGSGroupLayer, let childLayers = groupLayer.layers as? [AGSLayer], let childLayer = getLayer(from: childLayers, at: indexPath.row) else { return cell }
+        // Set label.
+        cell.textLabel?.text = formattedValue(of: childLayer.name)
+        
+        // Set the switch to on or off.
+        cell.layerVisibilitySwitch.isOn = childLayer.isVisible
+        
+        // To update the visibility of operational layers on switch toggle.
+        cell.delegate = self
+        
+        return cell
+    }
+    
+    // MARK: - UITableViewDelegate
+    
+    override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let headerView = tableView.dequeueReusableHeaderFooterView(withIdentifier: GroupLayersSectionView.reuseIdentifier) as? GroupLayersSectionView else { return nil }
+        
+        guard let layer = getLayer(from: layers, at: section) else { return nil }
+        // Set label.
+        headerView.layerNameLabel.text = formattedValue(of: layer.name)
+        
+        // Set the switch to on or off.
+        headerView.layerVisibilitySwitch?.isOn = layer.isVisible
+        if !layer.isVisible {
+            // Disable switch of child layers if the parent is not visible.
+            updateSwitchForRows(in: section, isEnabled: false)
+        }
+        
+        // To update the visibility of operational layers on switch toggle.
+        headerView.layerVisibilitySwitch?.tag = section
+        headerView.delegate = self
+        
+        return headerView
+    }
+    
+    // MARK: - GroupLayersCellDelegate
+    
+    func didToggleSwitch(_ cell: GroupLayersCell, isOn: Bool) {
+        guard let indexPath = tableView.indexPath(for: cell) else { return }
+        if let groupLayer = getLayer(from: layers, at: indexPath.section) as? AGSGroupLayer, let childLayers = groupLayer.layers as? [AGSLayer], let childLayer = getLayer(from: childLayers, at: indexPath.row) {
+            childLayer.isVisible = isOn
+        }
+    }
+    
+    // MARK: - GroupLayersSectionViewDelegate
+    
+    func didToggleSwitch(_ sectionView: GroupLayersSectionView, section: Int, isOn: Bool) {
+        let layer = getLayer(from: layers, at: section)
+        layer?.isVisible = isOn
+        updateSwitchForRows(in: section, isEnabled: isOn)
+    }
+    
+    // MARK: - Helper methods
+    
+    /// Ensures we cannot toggle visibility of child layers when the parent is turned off.
+    ///
+    /// - Parameters:
+    ///     - section: Section number of the table view.
+    ///     - enabled: Indicates if the switch should be enabled or disabled.
+    func updateSwitchForRows(in section: Int, isEnabled: Bool) {
+        guard let visibleRows = tableView.indexPathsForVisibleRows else { return }
+        let visibleRowsForSection = visibleRows.filter { $0.section == section }
```

Comment:
yes, I saw there was unnecessary check for section in line 106. I removed that. 

I think we do not need lazy filter because there is no chaining operations.

Reply:
> there is no chaining operations.

Yes and no. The `for` loop isn't a chained operation _per se_, but it has the same effect. It would be functionally equivalent rewritten as a chained operation:

```swift
visibleRows.filter { $0.section == section }
    .forEach { indexPath in
        if let cell = tableView.cellForRow(at: indexPath) as? GroupLayersCell {
            cell.layerVisibilitySwitch.isEnabled = isEnabled
        }
    }
```

In either implementation, there are two operations: 1) filter, 2) iterate. The `filter` operation creates a new array containing the filtered elements. `lazy.filter`, on the other hand, does not create a new array. It returns an instance of `LazyFilterCollection<Base>`, which is a view into the original collection. The filtering hasn't actually happened. It happens on demand as the the lazy filter collection is iterated. Does that make sense?

Obviously the benefit of using a lazy collection is proportional to the size of the original collection. The size of `visibleRows` is relatively small, so it probably wouldn't make any difference in this case.